db/migrate/20190422185437_add_location_table.rb

```diff
@@ -0,0 +1,27 @@
+class AddLocationTable < ActiveRecord::Migration[5.1]
+  def up
+    create_table :locations, force: true do |t|
+      t.string :name, default: "", null: false, comment: "Full display name, such as a complete address"
+      t.string :level_name, default: "", null: false, limit: 20, comment: "Level of specificity (country, state, subdivision, or city)"
+      t.string :country_name, default: "", null: false, limit: 100, comment: "Country (or equivalent) of this location if available"
+      t.string :state_name, default: "", null: false, limit: 100, comment: "State (or equivalent) of this location if available"
+      t.string :subdivision_name, default: "", null: false, limit: 100, comment: "Second-level administrative division (e.g. county/district/division/province/etc.) of this location if available"
+      t.string :city_name, default: "", null: false, limit: 100, comment: "City (or equivalent) of this location if available"
+
+      # Recommended settings for lat/lon
+      t.decimal :lat, precision: 10, scale: 6, comment: "The latitude of this location if available"
+      t.decimal :lon, precision: 10, scale: 6, comment: "The longitude of this location if available"
```

Comment:
I'm pretty sure the convention for longitude is `lng`  or `long` but not `lon`.

Reply:
I found some examples for both (OSM and LocationIQ use `lon` https://wiki.openstreetmap.org/wiki/Nominatim https://locationiq.com/docs). But Google/Mapbox/Leaflet use `lng` so I will switch and change everything to `lng`.